towhee/functional/mixins/ray.py

```diff
@@ -0,0 +1,84 @@
+from queue import Queue
+import asyncio
+import threading
+
+from towhee.utils.log import engine_log
+from towhee.functional.option import Option, Empty, _Reason
+from towhee.functional.mixins.parallel import EOS
+
+
+def _map_task_ray(unary_op):
```

Comment:
I suggest we have a common task class that represents such async computations. The task class also can support chunked execution which is able to reduce schedule overhead.